BUILD.md

```diff
@@ -0,0 +1,177 @@
+- Visual Studio >= 2015 (recommended)
+- GCC >= 4.8.1
+- Clang >= 3.3
+
+General information for OpenMVG CMake options
+- OpenMVG_BUILD_TESTS (ON/OFF(default))=> Build OpenMVG unit tests
+- OpenMVG_BUILD_EXAMPLES (ON/OFF(default))=> Build OpenMVG example applications.
+This does not affect binaries under 'software'
+
+
+General information for OpenMVG SfM pipelines
+
+- OpenMVG can export graphs as Graphviz .dot files and render them as SVG files. If you want this graph visualization feature, please install Graphviz.
+
+Linux compilation
+-----------------
+
+Install the required external libraries.
+```shell
+$ sudo apt-get install libpng-dev libjpeg-dev libtiff-dev libxxf86vm1 libxxf86vm-dev libxi-dev libxrandr-dev
+```
+If you want see the view graph svg logs, install Graphviz.
+```shell
+$ sudo apt-get install graphviz
+```
+
+Build OpenMVG.
+```shell
+$ git clone --recursive https://github.com/openMVG/openMVG.git
+$ cd openMVG
+$ ls
+ AUTHORS BUILD  docs  logo  README  src  ...
+$ cd ..
+$ mkdir openMVG_Build
+$ cd openMVG_Build
+```
+If you want to add unit tests and examples to the build, run
+```shell
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE -DOpenMVG_BUILD_TESTS=ON -DOpenMVG_BUILD_EXAMPLES=ON . ../openMVG/src/
+```
+otherwise
+```shell
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE . ../openMVG/src/
+```
+
+In order to use the MOSEK 6 back-end for the linear programming, OpenMVG checks that you have an up-to-date MOSEK licence, else OpenMVG MOSEK unit tests will fail.
+
+```shell
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE
+    -DMOSEK_SEARCH_HEADER="~/Documents/Lib/mosek/6/tools/platform/linux64x86/h"
+    -DMOSEK_SEARCH_LIB="~/Documents/Lib/mosek/6/tools/platform/linux64x86/bin"
+    . ../openMVG/src/
+```
+
+If you want to have an IDE openable project with Code::Blocks:
+```shell
+$ cmake -G "CodeBlocks - Unix Makefiles" -DCMAKE_BUILD_TYPE=RELEASE . ../openMVG/src/
+```
+
+Compile the project
+```shell
+$ make
+```
+or for a multi-core compilation. (Replace NBcore with the number of threads)
+```shell
+$ make -j NBcore
+```
+
+Run tests (if requested at CMake step)
+```shell
+$ make test
+```
+
+Have fun with the samples
+```shell
+$ cd openMVG_Samples
+```
+
+Windows compilation
+-------------------
+
+Checkout the project
+```shell
+$ git clone --recursive https://github.com/openMVG/openMVG.git
+```
+
+1. Open cmake-gui.
+2. Fill the source path with the src OpenMVG path.
+3. Fill the build path with a new directory.
+4. Select your Visual Studio IDE and click configure and then generate.
+5. Open the .sln solution created in your build directory.
+6. Change the target to Release.
+7. Compile the libraries and binaries samples.
+
+Mac compilation
+-------------------
+
+```shell
+$ git clone --recursive https://github.com/openMVG/openMVG.git
+$ cd openMVG
+$ ls
+ AUTHORS BUILD  docs  logo  README  src  ...
+$ cd ..
+$ mkdir openMVG_Build
+$ cd openMVG_Build
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE -G "Xcode" . ../openMVG/src/
```

Comment:
I know this was not in the previous BUILD file but since you make a good cleanup, I think we must say as in Linux compilation that we could do building in two different ways : 
1 ) using standard makefiles (i.e.: without the -G)
2) using Xcode 

in the case 1), compilation is made by invoking make , in case 2) compilation is as currently described.